### PR TITLE
Finished Game with score and recolectables

### PR DESCRIPTION
**Pull request by Rokziell**
Finished Game

### CODE REVIEW

**Comment by charly-palencia** [comment]
@Rokirito  pleases change your commit name. this is not the version 1, you should use a more descriptive name for your commit.

example `Implement Coins objects and GameOver functionality`

File: src/scenes/boot-scene.js [hunk 1]
```diff
@@ -1,40 +1,62 @@
...
   }
 
   create() {
     this.groundLayer = this.addGround();
     this.redBall = new Ball(this, 200, 200);
-    this.gameOverSound = this.sound.add("gameOver");
+    this.diamonds = this.addAllDiamonds();
+    this.gameOverSound = this.sound.add("gameOverSound");
+    this.keyR = this.input.keyboard.addKey("R");
+    this.score = 0;
```
Comment (charly-palencia) [inline src/scenes/boot-scene.js]: all this logic about the score should be move it into a method

File: src/scenes/boot-scene.js [hunk 2]
```diff
@@ -1,40 +1,62 @@
...
+
+    if(this.redBall.isDead && !this.isGameOver){
+      this.addGameOverMessage();
+      this.addGameOverSound();
+      this.addGameOver();
+      this.isGameOver = true;
+    }
+
+    if(this.redBall.isDead){
+      this.rToReset();
```
Comment (charly-palencia) [inline src/scenes/boot-scene.js]: what is a rToReset? please use a better name for this `resetGame` could be an option

File: src/scenes/boot-scene.js [hunk 3]
```diff
@@ -45,5 +67,64 @@ export default class BootScene extends Phaser.Scene {
...
+    });
+  }
+
+  collectDiamond(redBall, diamond){
+    this.score ++;
+    this.scoreText.setText("x " + this.score);
+    diamond.destroy();
+  }
+
+  addGameOver(){
```
Comment (charly-palencia) [inline src/scenes/boot-scene.js]: It's a good idea to split our method but you need to have for that. in this case `gameOver` is a functionality itself so you can create some method like `runGameOver` and then include the addGameOver, addGameOverSound and addFameOverMessage into it (also the `this.isGameOver = true;` in line 52). we don't need three different methods to understand what runGameOver method is doing

**Review by charly-palencia — CHANGES_REQUESTED** [review]

**Comment by charly-palencia** [comment]
@Rokirito  Include more feature details in this PR. for example what changes are you doing?

File: src/scenes/boot-scene.js (hunk 3 above)
Comment (Rokziell) [reply src/scenes/boot-scene.js]: Ok, I created a method called runGameOver that include addGameOver, addGameOverSound,  addGameOverMessage and the "this.isGameOver = true;" and put the method runGameOver inside the if(this.redBall is dead && !this.isGameOver)

File: src/scenes/boot-scene.js (hunk 2 above)
Comment (Rokziell) [reply src/scenes/boot-scene.js]: make the name's change to resetGame()

File: src/scenes/boot-scene.js (hunk 1 above)
Comment (Rokziell) [reply src/scenes/boot-scene.js]: all the logic about the score is inside a method called addScore()

**Pull request closed without merging**